libnavigator/src/androidTest/java/com/mapbox/navigation/navigator/internal/NavigatorTest.kt

```diff
@@ -0,0 +1,128 @@
+package com.mapbox.navigation.navigator.internal
+
+import androidx.annotation.RawRes
+import androidx.test.espresso.Espresso
+import androidx.test.platform.app.InstrumentationRegistry
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.navigator.internal.util.readRawFileText
+import com.mapbox.navigation.navigator.internal.util.runOnMainSync
+import com.mapbox.navigator.CacheFactory
+import com.mapbox.navigator.CacheHandle
+import com.mapbox.navigator.ConfigFactory
+import com.mapbox.navigator.ConfigHandle
+import com.mapbox.navigator.Navigator
+import com.mapbox.navigator.NavigatorConfig
+import com.mapbox.navigator.ProfileApplication
+import com.mapbox.navigator.ProfilePlatform
+import com.mapbox.navigator.Routes
+import com.mapbox.navigator.SettingsProfile
+import com.mapbox.navigator.TilesConfig
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class NavigatorTest {
+
+    private companion object {
+        @RawRes
+        private val DIRECTIONS_RESPONSE =
+            com.mapbox.navigation.navigator.test.R.raw.directions_response_two_leg_route
+    }
+
+    @Test(timeout = 10_000)
+    fun successfulSetRoutesToNN() {
+        runOnMainSync {
+            val navigator = provideNavigator()
+            val routes = provideDirectionsRouteAndRouteOptions()
+            navigator.setRoutes(
+                Routes(
+                    routes.mapToDirectionsResponse().toJson(),
+                    0,
+                    0,
+                    routes.first().routeOptions.toUrl("pk.**").toString()
+                )
+            ) { expected ->
+                assertTrue(expected.isValue)
+            }
+        }
+        Espresso.onIdle()
```

Comment:
I still don't understand the point of calling `Espresso.onIdle()` after `runOnMainSync`, which is blocking 🙂 
What does it wait for?

Reply:
I don't see any idling resource `Espresso.onIdle()` waits for. it seems like `Espresso.onIdle()` accidentally works for this case. It just waits for something and it just happens that this time is enough to set the route. I believe it't better to make sure that callback is called, see https://github.com/mapbox/mapbox-navigation-android/pull/5500/files#r820515193